pkgs/top-level/all-packages.nix

```diff
@@ -9248,9 +9248,8 @@ in
 
   buck = callPackage ../development/tools/build-managers/buck { };
 
-  buildkite-agent = buildkite-agent2;
-  buildkite-agent2 = callPackage ../development/tools/continuous-integration/buildkite-agent/2.x.nix { };
```

Comment:
I'm a fan of providing at least explanatory errors for removed attributes, something like
```nix
{
  buildkite-agent2 =
    throw "buildkite-agent2 is not supported anymore, use buildkite-agent instead";
}
```